lib/gtfs/components/gtfs-search.js

```diff
@@ -0,0 +1,255 @@
+// @flow
+
+import React, { Component, PropTypes } from 'react'
+import fetch from 'isomorphic-fetch'
+import { Glyphicon, Label } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import {Async} from 'react-select'
+
+import {GTFS_GRAPHQL_PREFIX} from '../../common/constants'
+import {getHeaders} from '../../common/util/util'
+import {routeSearch, stopSearch} from '../util/graphql'
+
+import type {Feed, GtfsRoute, GtfsStop, User} from '../../types'
+
+type Props = {
+  entities: Array<string>,
+  feeds: Array<Feed>,
+  filterByRoute: GtfsRoute,
+  filterByStop: GtfsStop,
+  onChange: any => void,
+  placeholder: ?string,
+  user: User,
+  value: string
+}
+
+type State = {
+  value: string
+}
+
+type Option = {
+  route?: GtfsRoute,
+  stop?: GtfsStop,
+  label: string,
+  value: string,
+  agency: Feed
+}
+
+/**
+ * A component build with react-select that queries the GraphQL endpoint for
+ * GTFS stops and routes using the user-inputted text.
+ */
+class GtfsSearch extends Component<Props, State> {
+  static propTypes = {
+    entities: PropTypes.array
+  }
+
+  state = {
+    value: this.props.value
+  }
+
+  static defaultProps = {
+    autoload: false, // prevent options from auto-loading, esp. when rendering multiple
+    entities: ['routes', 'stops'],
+    minimumInput: 1
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (this.state.value !== nextProps.value && typeof this.props.value !== 'undefined') {
+      this.setState({value: nextProps.value})
+    }
+  }
+
+  _renderOption (option: Option) {
+    return (
+      <span style={{ color: 'black' }}>
+        {option.stop
+          ? <Glyphicon glyph='map-marker' />
+          : <Glyphicon glyph='option-horizontal' />
+        }
+        {' '}{option.label}{' '}
+        <Label>
+          {option.agency && option.agency.name ? option.agency.name : ''}
+        </Label>
+      </span>
+    )
+  }
+
+  _sortOptions = (a: Option, b: Option, input: string) => {
+    const aLabel = a.label.toLowerCase()
```

Comment:
Move functions that don't depend on class variables outside the class.

Reply:
This doesn't depend on class variables, but it does depend on a class type `Option`.  Is it worth divorcing the type from the class just to move the sort function out?

Reply:
Yeah, you don't need to move it to another file, it can be a function within this file just not a method of the class.